### PR TITLE
Refactor/stats calculation

### PR DESCRIPTION
**Pull request by Uepsilon**
User- & Team-Statistic is updated:
-  whenever a match is created, updated or deleted
- from the total set of matches played by this user 

This prevents errors due to relational calculations. Might impact performance though

### CODE REVIEW

File: app/models/match.rb [hunk 1]
```diff
@@ -11,7 +11,12 @@ class Match < ActiveRecord::Base
   validates :loser_team, presence: true
   validate :team_players_validation
 
-  after_create :calculate_user_quotas
+  before_create :calc_difference
+  before_update :calc_difference
+
+  after_create :update_stats
+  after_update :update_stats
+  after_destroy :update_stats
```
Comment (DonSchado) [inline app/models/match.rb]: yay... delete 1 callback, to introduce 5

File: app/controllers/matches_controller.rb [hunk 2]
```diff
@@ -45,30 +45,26 @@ def edit
...
     @match.crawling = params[:crawling]
-    @match.calculate_user_quotas
     current_league.update_badges
     if @match.save
-      @match.update_team_streaks
-      redirect_to league_path(current_league), notice: "Satz gespeichert."
+      redirect_to league_path(current_league), notice: 'Satz gespeichert.'
     else
-      flash.now[:alert] = "Satz konnte nicht gespeichert werden."
+      flash.now[:alert] = 'Satz konnte nicht gespeichert werden.'
```
Comment (yoyostile) [inline app/controllers/matches_controller.rb]: Lass uns diesbezüglich einen Glaubenskrieg anzetteln!

**Comment by DonSchado** [comment]
:shipit:

File: app/controllers/matches_controller.rb (hunk 2 above)
Comment (Uepsilon) [inline app/controllers/matches_controller.rb]: Ich glaube nicht an "Glaubenskriege'

**Pull request merged**